src/irr/asset/CGLTFLoader.cpp

```diff
@@ -249,66 +273,48 @@ namespace irr
 						const E_FORMAT format = getFormat(glTFAccessor.componentType.value(), glTFAccessor.type.value());
 
 						auto& glTFbufferView = glTF.bufferViews[glTFAccessor.bufferView.value()];
-						const auto& relativeOffsetInBufferView = glTFAccessor.byteOffset.value();
-						const uint32_t bufferId = glTFbufferView.buffer.value();
+						const uint32_t& bufferBindingId = glTFbufferView.buffer.value();
+						const auto& globalOffsetInBufferBindingResource = glTFbufferView.byteOffset.value();
+						const auto& relativeOffsetInBufferViewAttribute = glTFAccessor.byteOffset.value();
 
 						typedef std::remove_reference<decltype(glTFbufferView)>::type SGLTFBufferView;
 
 						auto setBufferBinding = [&](uint32_t target) -> void
 						{
 							asset::SBufferBinding<ICPUBuffer> bufferBinding;
-							bufferBinding.offset = relativeOffsetInBufferView;
+							bufferBinding.offset = globalOffsetInBufferBindingResource;
 
-							cpuMeshBuffer->setIndexCount(glTFAccessor.count.value());
+							idReferenceBindingBuffers[bufferBindingId] = cpuBuffers[bufferBindingId];
+							bufferBinding.buffer = idReferenceBindingBuffers[bufferBindingId];
 
 							switch (target)
 							{
 								case SGLTFBufferView::SGLTFT_ARRAY_BUFFER:
 								{
-									idReferenceVertexBuffers[bufferId] = cpuBuffers[bufferId];
-									const uint32_t bindingId = idReferenceVertexBuffers.size();
-									bufferBinding.buffer = idReferenceVertexBuffers[bufferId];
-
-									cpuMeshBuffer->setIndexType(EIT_UNKNOWN);
-									cpuMeshBuffer->setVertexBufferBinding(std::move(bufferBinding), bindingId);
+									cpuMeshBuffer->setVertexBufferBinding(std::move(bufferBinding), bufferBindingId);
 
-									vertexInputParams.enabledBindingFlags |= core::createBitmask({ bindingId });
-									vertexInputParams.bindings[bindingId].inputRate = EVIR_PER_VERTEX;
-									vertexInputParams.bindings[bindingId].stride = 16; // TODO: check it
+									vertexInputParams.enabledBindingFlags |= core::createBitmask({ bufferBindingId });
+									vertexInputParams.bindings[bufferBindingId].inputRate = EVIR_PER_INSTANCE;
+									vertexInputParams.bindings[bufferBindingId].stride = 16; // TODO: check it
```

Comment:
stride would come from your glTF accessor

Reply:
corrected in following commit